routers/web/repo/actions/actions.go

```diff
@@ -122,6 +132,46 @@ func List(ctx *context.Context) {
 				}
 			}
 			workflows = append(workflows, workflow)
+
+			if len(workflowID) > 0 && ctx.Repo.IsAdmin() && workflow.Entry.Name() == workflowID {
+				events, err := actions.GetEventsFromContent(content)
+				if err != nil {
+					log.Warn("ignore check invalid workflow events %q: %v", workflow.Entry.Name(), err)
+				} else {
+					for _, event := range events {
+						if event.Name == webhook_module.HookEventWorkflowDispatch.Event() {
+							ctx.Data["AllowTriggerWorkflowDispatchEvent"] = true
+							ctx.Data["WorkflowDispatchConfig"] = event.WorkflowDispatch()
+
+							branchOpts := git_model.FindBranchOptions{
+								RepoID:          ctx.Repo.Repository.ID,
+								IsDeletedBranch: util.OptionalBoolFalse,
+								ListOptions: db.ListOptions{
+									ListAll: true,
+								},
+							}
+							branches, err := git_model.FindBranchNames(ctx, branchOpts)
+							if err != nil {
+								ctx.JSON(http.StatusInternalServerError, err)
+								return
+							}
+							// always put default branch on the top if it exists
+							if slices.Contains(branches, ctx.Repo.Repository.DefaultBranch) {
+								branches = util.SliceRemoveAll(branches, ctx.Repo.Repository.DefaultBranch)
+								branches = append([]string{ctx.Repo.Repository.DefaultBranch}, branches...)
+							}
+							ctx.Data["Branches"] = branches
+
+							tags, err := repo_model.GetTagNamesByRepoID(ctx, ctx.Repo.Repository.ID)
+							if err == nil {
+								ctx.Data["Tags"] = tags
+							}
+							break
+						}
+					}
+				}
+			}
+
```

Comment:
I think we only need provide `AllowTriggerWorkflowDispatchEvent` for the current workflow (you can move it to L190), and you are checking all of them here.
Then the `WorkflowDispatchConfig` can be a fetch request when user click `run workflow` button. We don't need to fetch it when loading the list page. This change can improve the performance.

Reply:
WorkflowDispatchConfig is parsed together in ParseRawOn. This operation is only processed once for the workflowId you select.
I guess this part of the problem is fetch branches and tags. I need some help to creat a branch selector component like `branch_dropdown.tmpl`. but the existing component is a menu,  <form> needs a dropdown selector. I'm not sure I have the ability to do this in Vue.

Reply:
Forgot to say, another benefit of moving them to L190 is that we can hide this notification when the workflow is disabled.

Reply:
Yes, I missed the Disabled check.